### PR TITLE
Add support for storing multiple auth tokens in fixie auth

### PR DESCRIPTION
**Pull request by petersalas**
This adds support for multiple auth tokens being stored in the Fixie CLI config. The config contains a default API token as well as a number of URL-specific tokens:

- When looking for a token we first check if there's a token specific to the current Fixie URL and fall back to the default token
- When updating the token we add the token to the host-specific tokens list _as well as the default token_. This means that for any previously-unseen URL, the last token recorded by `fixie auth` will be used.

### CODE REVIEW

**Review by mdwelsh — APPROVED** [review]
Nice!

File: fixieai/cli/auth/commands.py [hunk 1]
```diff
@@ -12,14 +13,14 @@ def validate_not_authed(ctx, param, force):
         # --force is set
         return
     try:
-        config = user_config.load_config()
-        if config.fixie_api_key is not None:
-            username = _get_username(config.fixie_api_key)
+        api_key = user_config.load_config().api_key
```
Comment (juberti) [inline fixieai/cli/auth/commands.py:16]: FYI that these are auth_tokens, not api keys - let's name them accordingly here

File: fixieai/cli/auth/user_config.py [hunk 2]
```diff
@@ -1,15 +1,51 @@
...
     fixie_api_key: Optional[str] = None
+    """The default API key, will be used if no URL-specific key matches."""
+
+    environment_api_keys: List[EnvironmentSpecificKey] = dataclasses.field(
+        default_factory=lambda: [],
+    )
+    """Environment-specific API keys."""
+
+    @property
+    def api_key(self) -> Optional[str]:
```
Comment (juberti) [inline fixieai/cli/auth/user_config.py:33]: see above on naming

File: fixieai/constants.py [hunk 3]
```diff
@@ -42,10 +40,11 @@ def fixie_api_key() -> str:
     if "FIXIE_API_KEY" in os.environ:
         return os.environ["FIXIE_API_KEY"]
     try:
-        with open(FIXIE_CONFIG_PATH) as fp:
-            api_key = yaml.safe_load(fp)["fixie_api_key"]
-            assert isinstance(api_key, str)
-            return api_key
+        from fixieai.cli.auth import user_config
+
+        api_key = user_config.load_config().api_key
```
Comment (juberti) [inline fixieai/constants.py:45]: see above on naming

**Review by juberti — APPROVED** [review]
LG but please adjust the naming to avoid confusion in the future

**Pull request merged**